### PR TITLE
Adapt to https://github.com/coq/coq/pull/19530

### PR DESCRIPTION
**Pull request by proux01**
Adapt to https://github.com/coq/coq/pull/19530
To be merged in sync with the upstream PR.

Most changes seem reasonable but example_055 may reveal an actual issue?

### CODE REVIEW

**Comment by JasonGross** [comment]
Yeah, example 55 seems indicative of an actual issue.  And I'd like a backwards compatible patch, the minimizer should continue to work with older versions of Coq.  Can we just use more regular expressions in the tests?

**Comment by proux01** [comment]
> Yeah, example 55 seems indicative of an actual issue.

Yes, I don't know what's happening there, would you mind having a look (I'd be happy to help if needed but I'm not sure I can do it myself, not being familiar with the minimizer, I don't precisely know what's expected and what's a correct or incorrect minimization result)

> And I'd like a backwards compatible patch, the minimizer should continue to work with older versions of Coq. Can we just use more regular expressions in the tests?

Other than this example_55 the remaining should be backward compatible

File: examples/run-example-008.py [hunk 1]
```diff
@@ -7,11 +7,14 @@
...
     'Require Coq.Arith.Arith.\nRequire Top.A.\nRequire Top.C.\nRequire Top.B.\nRequire Top.D.\n\nImport Top.D.\n\nFail Check A.mA.axA.\n',
     'Require Stdlib.Arith.Arith.\nRequire Top.A.\nRequire Top.C.\nRequire Top.B.\nRequire Top.D.\n\nImport Top.D.\n\nFail Check A.mA.axA.\n',
+    'Require Stdlib.Arith.All.Stdlib.Arith.Arith.\nRequire Top.A.\nRequire Top.C.\nRequire Top.B.\nRequire Top.D.\n\nImport Top.D.\n\nFail Check A.mA.axA.\n',
 )
 
 GET_EXPECT = {
     'Coq.Arith.Arith': None,
+    'Coq.Arith.All.Coq.Arith.Arith': None,
     'Top.A': 'Module Export Top_DOT_A.\nModule Export Top.\nModule A.\nImport Coq.Arith.Arith.\nModule mA.\n  Section secA.\n    Axiom axA : Set.\n  End secA.\nEnd mA.\n\nModule mA2.\nEnd mA2.\n\nEnd A.\n\nEnd Top.\n\nEnd Top_DOT_A.\n',
+    'Top.A': 'Module Export Top_DOT_A.\nModule Export Top.\nModule A.\nImport Coq.Arith.All.Coq.Arith.Arith.\nModule mA.\n  Section secA.\n    Axiom axA : Set.\n  End secA.\nEnd mA.\n\nModule mA2.\nEnd mA2.\n\nEnd A.\n\nEnd Top.\n\nEnd Top_DOT_A.\n',
```
Comment (JasonGross) [inline examples/run-example-008.py:17]: This is not backwards compatible.  Instead do
```suggestion
    'Top.A': [
        'Module Export Top_DOT_A.\nModule Export Top.\nModule A.\nImport Coq.Arith.Arith.\nModule mA.\n  Section secA.\n    Axiom axA : Set.\n  End secA.\nEnd mA.\n\nModule mA2.\nEnd mA2.\n\nEnd A.\n\nEnd Top.\n\nEnd Top_DOT_A.\n',
        'Module Export Top_DOT_A.\nModule Export Top.\nModule A.\nImport Coq.Arith.All.Coq.Arith.Arith.\nModule mA.\n  Section secA.\n    Axiom axA : Set.\n  End secA.\nEnd mA.\n\nModule mA2.\nEnd mA2.\n\nEnd A.\n\nEnd Top.\n\nEnd Top_DOT_A.\n'
        ],
```
and either make all of the others lists as well, or adapt the code below to iterate over possibilities when the expectation is not `None` and not a `str`.

**Comment by JasonGross** [comment]
> > Yeah, example 55 seems indicative of an actual issue.
> 
> Yes, I don't know what's happening there, would you mind having a look (I'd be happy to help if needed but I'm not sure I can do it myself, not being familiar with the minimizer, I don't precisely know what's expected and what's a correct or incorrect minimization result)

According to the [GitLab log](https://coq.gitlabpages.inria.fr/-/coq/-/jobs/4742945/artifacts/_build_ci/coq_tools/examples/example_055_make.log), the relevant issue is
```
WARNING: One or more physical paths match logical path Stdlib.Bool.Bool, but none of them exist: ../../../../_install_ci/lib/coq/theories/Bool/Bool.v
```
The minimizer reads `coqc -config` for the `COQLIB=` line and then appends `theories/` to find the location of the files, and then looks for files in that path according to the standard structure (reading from the .glob file).  So it seems that the .glob file on https://github.com/coq/coq/pull/19530 still believes the library is `Stdlib.Bool.Bool`, and `coqc -config` claims the stdlib lives in `../../../../_install_ci/lib/coq/theories/`, but the `.v` file is not there.  Does this seem like a plausible explanation of what's going wrong to you?  Can this issue be rectified on the Coq-side?

Source code in the minimizer:
https://github.com/JasonGross/coq-tools/blob/aa177d0640608ac59040d94e6403736bd3e549f6/coq_tools/find_bug.py#L1404-L1411
https://github.com/JasonGross/coq-tools/blob/aa177d0640608ac59040d94e6403736bd3e549f6/coq_tools/coq_version.py#L251-L256

**Comment by JasonGross** [comment]
Btw, probably `examples/Makefile` will need to be modified to move the relevant modified tests from `DEFAULT_TESTS` to `ONLY_IF_FROM_TESTS`, to avoid trying to build them on versions of Coq that are so old they do not support `From`.

File: examples/run-example-008.py (hunk 1 above)
Comment (proux01) [reply examples/run-example-008.py]: Thanks, fixed.

**Comment by proux01** [comment]
Thanks for the detailled explanation. So the file is now at `../../../../_install_ci/lib/coq/user-contrib/Stdlib/Bool/Bool.v`.
I fixed that, apparently it is now backward compatible, let CI confirm.

> Btw, probably examples/Makefile will need to be modified to move the relevant modified tests from DEFAULT_TESTS to ONLY_IF_FROM_TESTS, to avoid trying to build them on versions of Coq that are so old they do not support From.

Do you know when was `From` added? Apparently, it's so old that it's not even in the changelog.

**Comment by proux01** [comment]
@JasonGross thanks for the help. CI seems mostly happy, it's only complaining on 8.4 about `example_063` that you added earlier in the week. That error seems unrelated and that example shouldn't be in any way affected by the upstream PR. So as far as I'm concerned, this looks mergeable.
Of course, I can rebase if you want to confirm the green CI, but maybe not needed (and it is not green on master either, so probably hopeless).

**Comment by JasonGross** [comment]
Only example 63 fails, and I've been working on that one separately, so let's merge.

**Comment by JasonGross** [comment]
Thanks for your work preparing this!

**Comment by proux01** [comment]
Thanks for your help

**Pull request merged**